Warning reported for `null` passed to `params` array of nullable elements

Reports: `(8,15): warning CS8600: Cannot convert null to non-nullable reference.`
```
class C
{
    static void F(object x, params object?[] y)
    {
    }
    static void Main()
    {
        F("", null);
    }
}
```

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs
@@ -86630,15 +86630,15 @@ class C
 ";
             var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics(
-                // (16,15): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                // (16,25): warning CS8625: Cannot convert null literal to non-nullable reference type.
                 //         F(string.Empty, null); // 1
-                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(16, 15),
-                // (22,15): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(16, 25),
+                // (22,25): warning CS8625: Cannot convert null literal to non-nullable reference type.
                 //         G(string.Empty, null, string.Empty); // 2
-                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(22, 15),
-                // (27,30): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(22, 25),
+                // (27,40): warning CS8625: Cannot convert null literal to non-nullable reference type.
                 //         G(string.Empty, new object[] { null }); // 3
-                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(27, 30)
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(27, 40)
                 );
         }
     }

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs
@@ -86613,33 +86613,31 @@ class C
     {
         // Both calls here are invoked in normal form, not expanded
 
-        F("""", null); // 1
-        G("""", null);
+        F(string.Empty, null); // 1
+        G(string.Empty, null);
 
         // These are called with expanded form
 
-        F("""", null, """"); 
-        G("""", null, """"); // 2
+        F(string.Empty, null, string.Empty); 
+        G(string.Empty, null, string.Empty); // 2
 
         // Explicitly called with array
 
-        F("""", new object?[] { null }); 
-        G("""", new object[] { null }); // 3
-
-
+        F(string.Empty, new object?[] { null }); 
+        G(string.Empty, new object[] { null }); // 3
     }
 }
 ";
             var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics(
                 // (16,15): warning CS8625: Cannot convert null literal to non-nullable reference type.
-                //         F("", null); // 1
+                //         F(string.Empty, null); // 1
                 Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(16, 15),
                 // (22,15): warning CS8625: Cannot convert null literal to non-nullable reference type.
-                //         G("", null, ""); // 2
+                //         G(string.Empty, null, string.Empty); // 2
                 Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(22, 15),
                 // (27,30): warning CS8625: Cannot convert null literal to non-nullable reference type.
-                //         G("", new object[] { null }); // 3
+                //         G(string.Empty, new object[] { null }); // 3
                 Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(27, 30)
                 );
         }

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs
@@ -86593,5 +86593,55 @@ public class C
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "__refvalue(r, string?)").WithLocation(7, 13)
                 );
         }
+
+        [Fact, WorkItem(25375, "https://github.com/dotnet/roslyn/issues/25375")]
+        public void ParamsNullable_SingleNullParam()
+        {
+            var source =
+@"
+class C
+{
+    static void F(object x, params object?[] y)
+    {
+    }
+
+    static void G(object x, params object[]? y)
+    {
+    }
+
+    static void Main()
+    {
+        // Both calls here are invoked in normal form, not expanded
+
+        F("""", null); // 1
+        G("""", null);
+
+        // These are called with expanded form
+
+        F("""", null, """"); 
+        G("""", null, """"); // 2
+
+        // Explicitly called with array
+
+        F("""", new object?[] { null }); 
+        G("""", new object[] { null }); // 3
+
+
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (16,15): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         F("", null); // 1
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(16, 15),
+                // (22,15): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         G("", null, ""); // 2
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(22, 15),
+                // (27,30): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         G("", new object[] { null }); // 3
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(27, 30)
+                );
+        }
     }
 }

